src/backend/access/rmgrdesc/smgrdesc.c

```diff
@@ -27,7 +27,7 @@ smgr_desc(StringInfo buf, XLogReaderState *record)
 	{
 		xl_smgr_create *xlrec = (xl_smgr_create *) rec;
 		char	   *path = relpathperm(xlrec->rnode, xlrec->forkNum);
-		appendStringInfo(buf, "%s; smgr: %s", path, xlrec->impl == SMGR_MD ? "heap" : "ao");
+		appendStringInfo(buf, "%s; smgr: %s", path, xlrec->impl == SMGR_MD ? "heap" : (xlrec->impl == SMGR_AO ? "ao" : "unknown"));
```

Comment:
can we just have `smgr->smgr_name()` ?

Reply:
yeah.  maybe i will add a function like
`extern const char* smgr_get_name(SMgrImpl impl);`
It can handle custom smgr normally in the cloudberrydb process, but in tools such as pg_waldump, it can only be displayed as unknown